winrm/errors.go

```diff
@@ -0,0 +1,33 @@
+package winrm
+
+import (
+	"fmt"
+	"strings"
+)
+
+type ServerError struct {
```

Comment:
re-reading the original patch by @pecigonzalo , the idea behind adding the `err` in the `Client` is that if something went wrong at the http level, then that means this client won't be able to run any new commands, so it marks it as "errored" and aborts.

What we really need right now, is survive to `OperationTimeout` which will fix #48. I think we can get the other feature of the patch (ie mark a client as 'aborted' if there's an EOF), by just marking the `Client` aborted (which would require only one boolean field).

I'm going to keep only the first part of this patch, fix the `OperationTimeout` survival. Then I'll see how we can fix the original problem this patch wanted to fix (allow to reboot a VM or run `net stop`). I'll force re-push this PR as soon as I have something ready.